go/pkg/bertymessenger/service.go

```diff
@@ -526,6 +537,59 @@ func (svc *service) sendAccountUserInfo(groupPK string) error {
 	return nil
 }
 
+func (svc *service) streamInteraction(tx *dbWrapper, cid string, isNew bool) error {
+	if svc != nil && svc.dispatcher != nil {
+		eventInte, err := tx.getInteractionByCID(cid)
+		if err != nil {
+			return errcode.ErrDBRead.Wrap(err)
+		}
+
+		eventInte.Reactions, err = buildReactionsView(tx, cid)
+		if err != nil {
+			return errcode.ErrDBRead.Wrap(err)
+		}
+
+		if err := svc.dispatcher.StreamEvent(
+			mt.StreamEvent_TypeInteractionUpdated,
+			&mt.StreamEvent_InteractionUpdated{Interaction: eventInte},
+			isNew,
+		); err != nil {
+			return errcode.ErrMessengerStreamEvent.Wrap(err)
+		}
+	}
+	return nil
+}
+
+func buildReactionsView(tx *dbWrapper, cid string) ([]*mt.Interaction_ReactionView, error) {
+	reactions := ([]*mt.Reaction)(nil)
+	if err := tx.db.Where(&mt.Reaction{TargetCID: cid, State: true}).Find(&reactions).Error; err != nil {
```

Comment:
You might want to replace this query by a raw sql query which would be more efficient, it would look something like:

```sql
SELECT count(*) AS total_reactions, emoji, MAX(is_mine) > 0 AS has_mine
FROM reaction
WHERE target_cid = "QmA"
GROUP BY emoji
ORDER BY MIN(state_date) ASC```

https://www.db-fiddle.com/f/kLHpiAo5XQMXb7J12LGsUo/0

Reply:
thanks